File: cs/competitive_programming/leetcode_fulltimes/2022/december/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        hash1 = [0]*26
        hash2 = [0]*26

        for c in s:
            hash1[ord(c)-ord('a')] += 1
        for c in t:
            hash2[ord(c)-ord('a')] += 1

        return hash1==hash2

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s)!=len(t):
            return False

        d1 = collections.defaultdict(int)

        for c in s:
            d1[c] += 1
        for c in t:
            d1[c] -= 1

        for v in d1.values():
            if(v<0 or v>0):
                return False
        return True
        
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        out = [0]*26

        for c in s:
            k = ord(c)-ord('a')
            out[k] +=1
        for c in t:
            k = ord(c)-ord('a')
            out[k] -=1
        for v in out:
            if(v<0 or v>0):
                return False
        return True 

File: cs/competitive_programming/leetcode_fulltimes/2023/july/242_valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        int arr[26] = {0};

        int s1 = s.size();
        int s2 = t.size();
        int idx = 0;


        for(char c: s) {
            idx = (int)c - 'a';
            arr[idx] += 1;
        }

        for(char c: t) {
            idx = (int)c - 'a';
            arr[idx] -= 1;
        }


        for(int e: arr) {
            if(e!=0){
                return false;
            }
        }

        return true;

    }
};
